R/dataframe.R

```diff
@@ -157,6 +160,9 @@ as_data_frame.data.frame <- function(x, ...) {
 #' @export
 #' @rdname as_data_frame
 as_data_frame.list <- function(x, validate = TRUE, ...) {
+
+  x <- compact(x)
```

Comment:
Why is `compact()` here? Why should we always drop NULL columns in a data frame? What's the underlying principle?